app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,38 @@
+class CommentsController < ApplicationController
+  before_action :get_post, only: [:show, :edit, :create, :update, :destroy]
+  before_action :get_comment, only: [:edit, :update, :destroy]
+
+  def index
+  end
+
+  def create
+    @comment = @post.comments.create(comment_params)
+    redirect_to post_path(@post)
+  end
+
+  def edit
+  end
+
+  def update
+    @comment.update(comment_params)
+    redirect_to post_path(@post)
+  end
+
+  def destroy
+    @comment.destroy
+    redirect_to post_path(@post)
+  end
+
+  private
+    def comment_params
+      params.require(:comment).permit(:user, :comment, :post_id)
```

Comment:
Instead of allowing the user to insert a post Id, you can use the current one. In your update action for example:

```ruby
@comment.update(comment_params.merge(post: @post))
```